classes/message-handler.js

```diff
@@ -138,6 +138,22 @@ export default class MessageHandler extends EventEmitter{
             }, '*');
             break;
 
+        case 'switchRoom':
+            // Switch to this room
+            if (typeof args !== 'string') {
+                parent.postMessage({
+                    'status' : 'error',
+                    'message' : 'Invalid or missing arguments.'
+                }, '*');
+                return;
+            }
+            this.emit('switchRoom', args);
+            parent.postMessage({
+                'status' : 'success',
```

Comment:
do we have any way to handle this failing? once `switchRoom()` fires there is no going back?

Reply:
If `getRoom()` returns null then failure can be detected. However, the `MessageHandler` class only signals that it received a valid command. Further errors in getting the room would be handled inside the client.